src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -520,19 +520,8 @@ void SystemRequest::Run() {
     if (!file || !file->is_download_complete ||
         !file_system::MoveFile(app_full_file_path, file_dst_path)) {
       LOG4CXX_DEBUG(logger_, "Binary data not found.");
-
-      std::string origin_file_name;
-      if ((*message_)[strings::msg_params].keyExists(strings::file_name)) {
-        origin_file_name =
-            (*message_)[strings::msg_params][strings::file_name].asString();
-      }
-      if (!(mobile_apis::RequestType::HTTP == request_type &&
-            0 == origin_file_name.compare(kIVSU))) {
-        LOG4CXX_DEBUG(logger_, "Binary data required. Invalid data");
-        SendResponse(false, mobile_apis::Result::INVALID_DATA);
-        return;
-      }
-      LOG4CXX_DEBUG(logger_, "IVSU does not require binary data. Continue");
+      SendResponse(false, mobile_apis::Result::REJECTED);
```

Comment:
@LuxoftAKutsan This doesn't seem to fix the described problem, the original code seemed to fulfill that purpose (specifically for the file name `IVSU`, albeit the result code needed to change):
```
 -      std::string origin_file_name;
 -      if ((*message_)[strings::msg_params].keyExists(strings::file_name)) {
 -        origin_file_name =
 -            (*message_)[strings::msg_params][strings::file_name].asString();
 -      }
 -      if (!(mobile_apis::RequestType::HTTP == request_type &&
 -            0 == origin_file_name.compare(kIVSU))) {
 -        LOG4CXX_DEBUG(logger_, "Binary data required. Invalid data");
 -        SendResponse(false, mobile_apis::Result::INVALID_DATA);
 -        return;
 -      }
```
Was this supposed to be expanded to any case where binary data is not specified?

It also doesn't seem to do what you stated either (reject if a file path is specified but no binary data is included), since it doesn't check whether a file name is specified.

Reply:
@jacobkeeler the problem, described in #976 is that SDL replies with SUCCESS on SystemRequest with filename = IVSU and request type = HTTP, however SDL should reject such request. The original code was removed, because it is responsible exactly for this case.
In all other cases, when no binary data present, according to our requirements:
>In case mobile application sends valid SystemRequest with "HTTP" type, no binary data and file with "fileName" is not present in "SystemFilesPath" folder, SDL must respond with resultCode:REJECTED and success:false.

So INVALID_DATA result code is wrong here and it was replaced with correct one.
Also, code responsible for file_name parameter are at lines 463-468. In case when file name is not specified, default value of kSYNC constant will be taken.